ibis/backends/tests/test_vectorized_udf.py

```diff
@@ -83,6 +134,36 @@ def demean_struct(v, w):
     return v - v.mean(), w - w.mean()
 
 
+@analytic(
+    input_type=[dt.double, dt.double],
+    output_type=dt.Struct(['demean', 'demean_weight'], [dt.double, dt.double]),
+)
+def demean_struct_tuple_np_array(v, w):
+    assert isinstance(v, pd.Series)
+    assert isinstance(w, pd.Series)
+    return np.array(v - v.mean()), np.array(w - w.mean())
```

Comment:
Is it too hard to parameterized the return shape / type?

Reply:
You mean like define a single UDF and then have the test cases pass in the return type / shape (e.g. 'tuple', 'np.ndarray') and have the UDF figure out how to structure the output?

Reply:
More like generating different test fixture for each return shape

```
demean_struct_udfs = [
    create_demean_struct_udf(output_shape=np.array)
    create_demean_struct_udf(output_shape=tuple)
    create_demean_struct_udf(output_shape=pd.Series)
    create_demean_struct_udf(output_shape=pd.DataFrame)
]
```

Reply:
Or sth like 

```
demean_struct_udfs = [
create_demean_struct_udf(output_shape=lambda v1, v2: (v1.values, v2.values)) # tuple of numpy array,
create_demean_struct_udf(output_shape=lambda v1, v2: pd.DataFrame({'v1': v1, 'v2': v1})) # dataframe with matching names 
]
```

Reply:
After trying this out I think there are pros and cons of this approach - the pros are that the list (e.g. `demean_struct_udfs`) makes it very clear what is the set of UDFs and their different return types that we're testing, and would be easy to add to in the future. The con is that in order for `create_demean_struct_udf` to return a function that is decorated with the UDF annotation and then wrapped in the lambda that formats the result, we change the signature of the original UDF to be a regular function, which then affects other tests that reuse these UDFs - e.g. they also have to call the `create` function to use them.